### PR TITLE
Reverted back to main

### PR DESCRIPTION
**Pull request by Shawn-Jemmott**
@ypkang had to delete my fork repo and fork back because these merge conflicts was taking rather long to resolve I followed almost every article. But nevertheless got the job done.

### CODE REVIEW

File: examples/canoniCAI/data/ner_gen [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import json
```
Comment (ypkang) [inline examples/canoniCAI/data/ner_gen]: What is this file for? Do we need to commit this?

File: examples/canoniCAI/data/ner_train.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (ypkang) [inline examples/canoniCAI/data/ner_train.json]: The indentation of this file is different from rest of the json files.

File: examples/canoniCAI/documentation/how to add a competency in canoniCAI.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
+# CanoniCAI
```
Comment (ypkang) [inline examples/canoniCAI/documentation/how to add a competency in canoniCAI.md]: Let's not include space in file names. So instead of `how to add a competency in canoniCAI.md` it can just be `add_competency.md`

File: examples/canoniCAI/documentation/how to add and remove data from faq state.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# How to add and remove data from faq state
```
Comment (ypkang) [inline examples/canoniCAI/documentation/how to add and remove data from faq state.md]: Same here with the file name. See previous comment above.

File: jaseci_kit/jaseci_kit/modules/object_detection/api.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
+from fastapi import FastAPI, Form, File, UploadFile
```
Comment (ypkang) [inline jaseci_kit/jaseci_kit/modules/object_detection/api.py]: Can you separate the yolo model into its own PR?

File: jaseci_kit/jaseci_kit/modules/object_detection/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,46 @@
+# YOLOv5 requirements
+# Usage: pip install -r requirements.txt
+import sys
+sys.path.insert(0, './yolov5')
```
Comment (ypkang) [inline jaseci_kit/jaseci_kit/modules/object_detection/requirements.txt]: Is this necessary? Does it even work inside requirements.txt?

File: jaseci_kit/jaseci_kit/modules/object_detection/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+# Export --------------------------------------
+# coremltools>=4.1  # CoreML export
+# onnx>=1.9.0  # ONNX export
+# onnx-simplifier>=0.4.1  # ONNX simplifier
+# nvidia-pyindex  # TensorRT export
+# nvidia-tensorrt  # TensorRT export
+# scikit-learn==0.19.2  # CoreML quantization
+# tensorflow>=2.4.1  # TFLite export
+# tensorflowjs>=3.9.0  # TF.js export
+# openvino-dev  # OpenVINO export
```
Comment (ypkang) [inline jaseci_kit/jaseci_kit/modules/object_detection/requirements.txt]: Why are these packages commented out? Are they not needed?

**Review by ypkang — CHANGES_REQUESTED** [review]
See comments in line.
Also, i see there is a git submodule added for yolov5. Is that intentional?

**Comment by ypkang** [comment]
@Shawn-Jemmott check comments here.

File: examples/canoniCAI/data/ner_gen (hunk 1 above)
Comment (Shawn-Jemmott) [reply examples/canoniCAI/data/ner_gen]: this used to auto ner json template based on the ner training data. Because we have no frontend which will tell us the start index and end index of a slot and generate it to a json. I used it there to help me generate a simple json for single slot cases for utterances in the training set.

File: examples/canoniCAI/data/ner_train.json (hunk 2 above)
Comment (Shawn-Jemmott) [reply examples/canoniCAI/data/ner_train.json]: This is auto generated from the ner_gen file because it is quite difficult to train ner manually.

File: examples/canoniCAI/documentation/how to add a competency in canoniCAI.md (hunk 3 above)
Comment (Shawn-Jemmott) [reply examples/canoniCAI/documentation/how to add a competency in canoniCAI.md]: will fix this now

File: jaseci_kit/jaseci_kit/modules/object_detection/requirements.txt (hunk 6 above)
Comment (Shawn-Jemmott) [reply jaseci_kit/jaseci_kit/modules/object_detection/requirements.txt]: This is what I got from the documentation from yolo. requirements.txt file is only for pip install and not to insert sys.oath

File: jaseci_kit/jaseci_kit/modules/object_detection/requirements.txt (hunk 7 above)
Comment (Shawn-Jemmott) [reply jaseci_kit/jaseci_kit/modules/object_detection/requirements.txt]: This was directly from yoloV5 clone branch documetation. I did not touch these because its probably important if we want certain features in future for object detection

File: jaseci_kit/jaseci_kit/modules/object_detection/api.py (hunk 5 above)
Comment (Shawn-Jemmott) [reply jaseci_kit/jaseci_kit/modules/object_detection/api.py]: I was trying this but it kept updating the previous PR and if I had to create a different branch the previous PR is still attached which causes problems

File: examples/canoniCAI/readme.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
+## Getting Started CanoniCAI
+
+##### Clone CanonCAI repo using: https://github.com/marsninja/jac_convAI
```
Comment (ypkang) [inline examples/canoniCAI/readme.md]: This needs to be updated

File: examples/canoniCAI/data/ner_gen (hunk 1 above)
Comment (ypkang) [reply examples/canoniCAI/data/ner_gen]: Rename this to `ner_gen.py` and add at the top of this file a comment block explaining what this file is and how to use it.

"""
This file is for blah blah blah.
You can run it with Blah blah balh
"""

File: jaseci_kit/jaseci_kit/modules/object_detection/api.py (hunk 5 above)
Comment (ypkang) [reply jaseci_kit/jaseci_kit/modules/object_detection/api.py]: Make a new branch for yolov5 and open a new PR.

File: examples/canoniCAI/readme.md (hunk 8 above)
Comment (ypkang) [reply examples/canoniCAI/readme.md]: Getting started with CanoniCAI
